tests/std/tests/GH_002030_asan_annotate_string/test.cpp

```diff
@@ -0,0 +1,1815 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+// REQUIRES: asan, x64 || x86

```

Comment:
After removing the compiler bug workaround, this test is now failing for x86 (but passing for x64 - sorry about that, I validated x64 only before pushing). I'm unsure whether this is another compiler bug, or a library bug.

There appear to be 2 forms of failures:

1. Insufficient alignment:
   ```
   ==3212==ERROR: AddressSanitizer: bad parameters to __sanitizer_annotate_contiguous_container:
         beg     : 0x021be724
         end     : 0x021be734
         old_mid : 0x021be734
         new_mid : 0x021be728
   ==3212==ERROR: beg is not aligned by 8
   ```
   This looks like it might be a library bug - those pointers are definitely not aligned.

2. stack-buffer-underflow:
   ```
   ==9892==ERROR: AddressSanitizer: stack-buffer-underflow on address 0x0022fac4 at pc 0x00405868 bp 0x0022fa94 sp 0x0022fa88
   READ of size 4 at 0x0022fac4 thread T0
   [...]
   Address 0x0022fac4 is located in stack of thread T0 at offset 4 in frame
       #0 0x4080bf  (D:\build\tests\std\tests\GH_002030_asan_annotate_string\Output\44\GH_002030_asan_annotate_string.exe+0x4280bf)
   
     This frame has 2 object(s):
       [16, 17) '_Proxy' (line 2702)
       [32, 36) '_New_ptr' (line 2744)
   HINT: this may be a false positive if your program uses some custom stack unwind mechanism, swapcontext or vfork
         (longjmp, SEH and C++ exceptions *are* supported)
   SUMMARY: AddressSanitizer: stack-buffer-underflow (D:\build\tests\std\tests\GH_002030_asan_annotate_string\Output\44\GH_002030_asan_annotate_string.exe+0x425867)
   Shadow bytes around the buggy address:
   [...]
   ```
   Not sure what's happening here.